lib/errors.c

```diff
@@ -536,18 +536,29 @@ guestfs_int_external_command_failed (guestfs_h *g, int status,
 {
   const size_t len = 80 + strlen (cmd_name);
   CLEANUP_FREE char *status_string = safe_malloc (g, len);
+  CLEANUP_FREE char *extra_clean = NULL;
 
   guestfs_int_exit_status_to_string (status, cmd_name, status_string, len);
 
+  if (extra) {
+    extra_clean = safe_strdup(g, extra);
```

Comment:
You can use `guestfs_int_replace_string` here.  It can return `NULL` which you have to deal with which is annoying, but other than that it could be a bit easier.